lib/msf/core/modules/metadata/search.rb

```diff
@@ -183,13 +185,15 @@ def is_match(params, module_metadata, regex_cache)
             when 'arch'
               match = [keyword, search_term] if module_metadata.arch =~ regex
             when 'cve'
-              match = [keyword, search_term] if module_metadata.references.any? { |ref| ref =~ /^cve\-/i and ref =~ regex }
+              match = [keyword, search_term] if module_metadata.references.any? { |ref| ref.downcase.start_with?('cve-') && ref =~ regex }
+            when 'att&ck', 'attack'
+              match = [keyword, search_term] if module_metadata.references.any? { |ref| ref.downcase.start_with?('att&ck-') && ref =~ regex }
```

Comment:
So it looks like the other keywords are a bit more fuzzy than I would have expected. If we're searching by a specific ID, I'd expect it to find all that have either that exact ID, or a sub ID from that exact ID.

To achieve this, I updated this section here and it's now working as I'd expect.

```suggestion
              regex = Regexp.new("\\A#{Regexp.escape(search_term)}(\\.\\d+)*\\Z", Regexp::IGNORECASE)
              match = [keyword, search_term] if module_metadata.references.any? { |ref| ref.downcase.start_with?('att&ck-') && ref.downcase.delete_prefix('att&ck-') =~ regex }
```

Before, if you searched for `att&ck:t1` you'd get everything with a ID starting with `T1`. Now you'll only get ones that match T1 exactly or a subcategory (e.g. `T1.##`).

For demonstration, search for `att&ck:T1059`, `att&ck:T1059.001`, `att&ck:t1059.003` and see different results for each. Then search for `att&ck:T1` and see nothing.